src/include/rbd/librbd.hpp

```diff
@@ -293,6 +293,7 @@ class CEPH_RBD_API Image
   int snap_rename(const char *srcname, const char *dstname);
   int snap_get_limit(uint64_t *limit);
   int snap_set_limit(uint64_t limit);
+  int snap_get_timestamp(uint64_t snap_id, std::string *timestamp);
```

Comment:
The API should provide either a ```timespec *``` out parameter or a ```double *``` out parameter, not a string. You also need to add this to the C and Python APIs